sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2065,18 +2065,28 @@ object EliminateUnions extends Rule[LogicalPlan] {
  */
 object CleanupAliases extends Rule[LogicalPlan] {
   private def trimAliases(e: Expression): Expression = {
-    var stop = false
     e.transformDown {
-      // CreateStruct is a special case, we need to retain its top level Aliases as they decide the
-      // name of StructField. We also need to stop transform down this expression, or the Aliases
-      // under CreateStruct will be mistakenly trimmed.
-      case c: CreateStruct if !stop =>
-        stop = true
-        c.copy(children = c.children.map(trimNonTopLevelAliases))
-      case c: CreateStructUnsafe if !stop =>
-        stop = true
-        c.copy(children = c.children.map(trimNonTopLevelAliases))
-      case Alias(child, _) if !stop => child
+      /**
+        *  [[CreateStruct]] is a special case as it uses its top level Aliases to decide the
+        *  name of StructField.
+        *  we tackle this by replacing [[CreateStruct]] with [[CreateNamedStruct]]
+        *  which encodes the field names as child literals.
+        */
+      case c @ CreateStruct(children) =>
```

Comment:
can we remove `CreateStruct` expression entirely if `CreateNamedStruct` covers all its usage?